src/App.js

```diff
@@ -1,5 +1,133 @@
+import { Console, Random } from '@woowacourse/mission-utils';
+
 class App {
-  async play() {}
+  async play() {
+    Console.print('숫자 야구 게임을 시작합니다.');
+
+    while (true) {
+      await this.gameFlow();
+      const userRestartInput = await this.getUserRestartInput();
+      if (userRestartInput !== 1) {
+        break;
+      }
+    }
+  }
+  // 게임 실행 흐름을 책임지는 함수
+  async gameFlow() {
+    const COMPUTER_ANSWER = this.createComputerAnswer();
+    let isUserWon = false;
+
+    while (!isUserWon) {
+      isUserWon = await this.checkUserGuess(COMPUTER_ANSWER);
+    }
+    return;
+  }
+
+  // 사용자 입력을 받는 getUserInput
+  async getUserInput() {
+    try {
+      const answer = await Console.readLineAsync('숫자를 입력해주세요 : ');
+
+      const userInput = answer.split('');
+
+      if (
+        userInput.length !== 3 ||
+        userInput.some(char => isNaN(parseInt(char, 10))) ||
+        new Set(userInput).size !== 3
+      ) {
+        throw new Error(
+          '[ERROR] 1에서 9 사이의 서로 다른 세 숫자를 입력해 주세요.'
+        );
+      }
+
+      return userInput.map(char => parseInt(char, 10));
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  // 랜덤한 컴퓨터 정답을 생성하는 함수
+  createComputerAnswer = () => {
+    let computerAnswer = [];
+    while (computerAnswer.length < 3) {
+      let randomNum = Random.pickNumberInRange(1, 9);
+
+      if (!computerAnswer.includes(randomNum)) {
+        computerAnswer.push(randomNum);
+      }
+    }
+    return computerAnswer;
+  };
+
+  // 낫싱을 확인하는 함수
+  isNothing = (userGuess, computerAnswer) => {
+    return userGuess.filter(x => computerAnswer.includes(x)).length === 0
+      ? true
+      : false;
+  };
+
+  // 스트라이크와 볼을 확인하는 함수
+  checkStrikeOrBall = (userGuess, computerAnswer) => {
+    let strikeCount = 0,
+      ballCount = 0;
+
+    for (let i = 0; i < userGuess.length; i++) {
+      if (userGuess[i] === computerAnswer[i]) {
+        strikeCount += 1;
+      } else if (computerAnswer.includes(userGuess[i])) {
+        ballCount += 1;
+      }
+    }
+
+    return { strikeCount, ballCount };
+  };
+
+  // 사용자의 추측을 출력하는 함수
+  printGuessResult = guessResult => {
+    if (guessResult.strikeCount === 3) {
+      Console.print('3스트라이크\n3개의 숫자를 모두 맞히셨습니다! 게임 종료');
+    } else if (guessResult.strikeCount === 0) {
+      Console.print(`${guessResult.ballCount}볼`);
+    } else if (guessResult.ballCount === 0) {
+      Console.print(`${guessResult.strikeCount}스트라이크`);
+    } else
+      Console.print(
+        `${guessResult.ballCount}볼 ${guessResult.strikeCount}스트라이크`
+      );
```

Comment:
조건문의 순서를 바꾸면 if return 으로도 대체될 수 있을 것 같다는 생각이 들어요!
문자열을 return 해주는 부분이 조금 다르긴 하지만,, 저의 경우에는 이렇게 구현했어요!
```
getHintString(ball, strike) {
    if (ball === 0 && strike === 0) {
      return '낫싱';
    }
    if (ball === 0) {
      return `${strike}스트라이크`;
    }
    if (strike === 0) {
      return `${ball}볼`;
    }

    return `${ball}볼 ${strike}스트라이크`;
  }
```


그리고 비구조화 할당을 사용해보면 어떨까요? 구조화대입을 이용하는 것으로 
프로퍼티를 위한 임시적인 참조의 작성을 줄이고, 가독성을 높일 수 있을 것 같아요!

```suggestion
  printGuessResult ={strikeCount, ballCount} => {
    if (strikeCount === 3) {
      Console.print('3스트라이크\n3개의 숫자를 모두 맞히셨습니다! 게임 종료');
    } else if (strikeCount === 0) {
      Console.print(`${ballCount}볼`);
    } else if (guessResult.ballCount === 0) {
      Console.print(`${strikeCount}스트라이크`);
    } else
      Console.print(
        `${ballCount}볼 ${strikeCount}스트라이크`
      );
```

![image](https://github.com/woowacourse-precourse/javascript-baseball-6/assets/86418674/d7f8d4a0-e8df-40d2-9d4d-cba02463f769)

Reply:
이 부분은 조금 더 읽어봐야 되겠네요. 자세한 설명 감사합니다!